_pytest/python.py

```diff
@@ -928,7 +928,12 @@ def _find_parametrized_scope(argnames, arg2fixturedefs, indirect):
 
 def _idval(val, argname, idx, idfn, config=None):
     if idfn:
-        s = idfn(val)
+        s = None
+        try:
+            s = idfn(val)
+        except Exception:
+            import warnings
+            warnings.warn("Raised while trying to determine id of parameter %s at position %d" % (argname, idx))
```

Comment:
I think this will fit well when we integrate pytest-warnings into `3.1`. 😁 

Could you please add that this will turn into an error on pytest-4.0, something like:

```python
msg = "Raised while trying to determine id of parameter %s at position %d" % (argname, idx)
msg += '\nNote that this will turn into an error in pytest-4.0.'
warnings.warn(msg)
```